plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -87,6 +87,61 @@ func (z *Zfs) gatherPoolStats(acc telegraf.Accumulator) (string, error) {
 	return strings.Join(pools, "::"), nil
 }
 
+func (z *Zfs) gatherDatasetStats(acc telegraf.Accumulator) (string, error) {
+
+	lines, err := z.zdataset()
```

Comment:
You could define the information to gather here and pass it on to the `zdataset()` call like:
```suggestion
	properties := []string {"name", "avail", "used", "usedsnap", "usedds"}
	
	lines, err := z.zdataset(properties)
```